### DISCUSSION TITLE
Make sure the Cursor is initialized correctly before accessing data from it.

### DISCUSSION THREAD

**Discussion by mochadwi** [google/modernstorage#98] [Q&A]
```
Fatal Exception: java.lang.IllegalStateException: Couldn't read row 0, col 3 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
       at android.database.CursorWindow.nativeGetLong(CursorWindow.java)
```

Is there anything to setup/do to prevent the above happened?

We're trying to calls the below snippet in: `registerForActivityResult`

```kotlin
context.fileSystem.metadata(path)
        .extra(MetadataExtras.MimeType::class)?.value
```

Full snippet:

```kotlin
    private val openDocument =
        componentActivity.registerForActivityResult(ActivityResultContracts.OpenDocument()) { uri ->
            if (uri != null) {
                onOpenDocumentResult(
                    uri,
                    componentActivity.fileSystem.metadata(uri.toOkioPath())
        .extra(MetadataExtras.MimeType::class)?.value,
                    componentActivity.fileSystem.metadata(uri.toOkioPath())
        .extra(MetadataExtras.DisplayName::class)?.value,
                )
            } else {
                // do nothing
            }
        }

```

**Comment by yrezgui** [comment]
Do you have the URI you were trying to fetch metadata from? Also any information on the device would be helpful (Android SDK version, target SDK, model, etc.)

**Reply by mochadwi** [reply]
Unfortunately no, the above crash shown from Crashlytics and we are not logging the Uri atm. 

But after a while checking this seems only happened when user open/pick file from Google Photos or Google Drive

This is what the Uri looks like: https://github.com/Blankj/AndroidUtilCode/issues/1439 (content://com.google.android.apps.docs.storage/document/acc%3D3%3Bdoc%3Dencoded%3DGp3qNtSd17JyIvWWjN4kiMe66gjReS6bFjH9yJoRGAp7CAxPF3tSqyiUVZQX6AWznm4%3D)

Does modernstorage currently not support either Photos or Gdrive(?)

**Reply by mochadwi** [reply]
<img width="1096" alt="Screen Shot 2022-05-18 at 3 28 04 PM" src="https://user-images.githubusercontent.com/7057959/168993992-d0258bb9-1331-47a4-831d-3854fbc5df77.png">

**Reply by mochadwi** [reply]
> Do you have the URI you were trying to fetch metadata from? Also any information on the device would be helpful (Android SDK version, target SDK, model, etc.)

Target SDK: 30
Android SDK: (I'm not sure to obtain this info from where, does it compileSdk? or API level in the user's phone?)

**Reply by mochadwi** [reply]
this still happened tho' :(

**Reply by mochadwi** [reply]
Any suggestion? 🙏
